w3af/plugins/infrastructure/hmap.py

```diff
@@ -66,10 +66,15 @@ def discover(self, fuzzable_request):
             port = 80
             ssl = False
 
-        # Override the defaults
-        if server.count(':'):
-            port = int(server.split(':')[1])
-            server = server.split(':')[0]
+        # Fix for IPv6 address - remove brackets
```

Comment:
This logic doesn't work when the port is specified!

https://[2001:db8:85a3:8d3:1319:8a2e:370:7348]:443/